File: plus-minus/plus-minus.js
// https://www.hackerrank.com/challenges/plus-minus/problem

const plusMinus = integers => {
  let plusCounter = 0;
  let minusCounter = 0;
  let zeroCounter = 0;

  integers.forEach(integer => {
    switch (true) {
      case integer > 0:
        return plusCounter++;
      case integer < 0:
        return minusCounter++;
      default:
        return zeroCounter++;
    }
  });

  const { length } = integers;
  [plusCounter, minusCounter, zeroCounter].forEach(counter => {
    console.log(counter / length);
  });
};

File: plus-minus/plus-minus.clj
(defn counters [ns]
  (reduce
   (fn [accum n]
     (let [{plus :plus, minus :minus, zero :zero} accum]
       (cond
         (pos? n) (merge accum {:plus (inc plus)})
         (neg? n) (merge accum {:minus (inc minus)})
         :else (merge accum {:zero (inc zero)}))))
   {:plus 0 :minus 0 :zero 0}
   ns))

(defn plusMinus [ns]
  (let [{plus :plus, minus :minus, zero :zero} (counters ns)
        length (count ns)]
    (doseq [counter [plus minus zero]] (println (double (/ counter length))))))